File: app/src/main/java/com/itmoldova/detail/DetailContract.java
package com.itmoldova.detail;

import android.view.View;

import com.itmoldova.model.Item;

import java.util.List;

/**
 * Author vgrec, on 24.07.16.
 */
public class DetailContract {

    interface View {
        void showArticleDetail(List<android.view.View> views);

        void showRelatedArticles(android.view.View relatedArticles);

        void showTitle(String title);

        void showHeaderImage(String url);

        void hideHeaderImage();
    }

    interface Presenter {
        void loadArticleDetail(Item item);

        void loadRelatedArticles(List<Item> items, Item item);

        List<String> extractPhotoUrlsFromArticle();
    }
}

File: app/src/main/java/com/itmoldova/detail/DetailContract.kt
package com.itmoldova.detail

import com.itmoldova.model.Article

class DetailContract {

    interface View {
        fun showArticleDetail(content: String)

        fun showRelatedArticles(relatedArticles: List<Article>)

        fun showTitle(title: String)

        fun showHeaderImage(url: String)

        fun hideHeaderImage()

        fun updateFavoriteIcon(iconResId: Int)
    }

    interface Presenter {
        fun loadArticleDetail(article: Article)

        fun loadRelatedArticles(topArticles: List<Article>, article: Article)

        fun addOrRemoveFromBookmarks(isAlreadyAdded: Boolean, article: Article)

        fun setProperBookmarkIcon(article: Article)

        fun cancel()
    }
}
